phoenix-core/src/it/java/org/apache/phoenix/end2end/HashJoinIT.java

```diff
@@ -3453,4 +3496,27 @@ public void testJoinWithSetMaxRows() throws Exception {
             conn.close();
         }
     }
+    
+    public static class InvalidateHashCacheRandomly extends SimpleRegionObserver {
```

Comment:
These join tests run on the same cluster as all other tests derived from ParallelStatsDisabledIT. Rather than having a random test failure, let's move this to a dedicated test derived from BaseUniqueNamesOwnClusterIT and have it consistently fail.